python-test/features/steps/control_plane_policies.py

```diff
@@ -42,6 +50,19 @@ def new_policy(context):
     check_policies(context)
 
 
+@step('the container logs that were output after all policies have been applied contains the message "{'
+      'text_to_match}" referred to each policy within {time_to_wait} seconds')
+def check_agent_logs_for_policies(context, text_to_match, time_to_wait):
+    check_agent_log_for_policies(text_to_match, time_to_wait, context.container_id, context.list_agent_policies_id,
+                                 context.dataset_applied_timestamp)
+
+
+@step('the container logs contains the message "{text_to_match}" referred to each policy within {'
+      'time_to_wait} seconds')
+def check_agent_logs_for_policies(context, text_to_match, time_to_wait):
```

Comment:
There are 2 functions with the same signature (same name `check_agent_logs_for_policies` and same input parameters). Isn't this going to cause an issue when python executes the code?

Reply:
Since the function does the same thing in both places, behave does not generate redundancy if the more specific function (with extra parameters) is defined first, but in any case, I agree that it is better to keep the pattern. We will keep different names for all functions.